search-endpoint.md

```diff
@@ -0,0 +1,397 @@
+# Discovery Search Endpoints
+[Back to the list of all defined endpoints](endpoints.md)
+
+## Search Endpoint
+**/api/discover/search**   
+
+This endpoint provides the functionality of the Discovery search screen: http://demo.dspace.org/xmlui/discover.
+
+Provide access to the Discovery search system (SOLR based). This endpoint returns the list of available DSpace object (DSO) types that can be searched:
+* all: Execute a query over all available DSO types.
+* items: Only search the DSpace items.
+* communities: Search within the DSpace community records.
+* collections: Limit the search to DSpace collection records.
+
+Example: TODO
+
+### Search with a given DSO type
+**/api/discover/search/<:dso-type>**
```

Comment:
The "dso-type" is part of the URL because the DSpace REST API framework forces me to build URLs in the form of `/<category>/<model>/<id>/<rel>` (as defined [here](https://github.com/DSpace/DSpace/blob/master/dspace-spring-rest/src/main/java/org/dspace/app/rest/RestResourceController.java#L60) and [here](https://github.com/DSpace/DSpace/blob/master/dspace-spring-rest/src/main/java/org/dspace/app/rest/RestResourceController.java#L129)). In this case, "dso-type" would be the "id" of the search. I don't really like this but I don't see another way currently with the framework we have.

Restricting the endpoint to `/<category>/<model>` ("/discover/search" in my case) would require modifications to `org.dspace.app.rest.RestResourceController` I think. @abollini, do you have any recommendations on how to do that?